File: frontend/apps/tenant-admin-web/src/api/menu.ts
import { request } from '@ark-iam/api'
import type { MenuTreeResp, TenantAppItem } from '@ark-iam/types'

/** 获取当前租户的菜单树（租户自服务控制台侧边栏动态菜单） */
export const getMyMenuTree = () => request.get<any, MenuTreeResp>('/tenant/menus/tree')

/** 获取当前租户订阅的应用列表（角色归属 / 菜单授权的应用选项） */
export const getTenantApps = () => request.get<any, { list: TenantAppItem[] }>('/tenant/apps')

File: backend/apps/tenantadmin/internal/controller/ctrtenant/menu.go
package ctrtenant

import (
	"github.com/gin-gonic/gin"
	"github.com/morehao/ark-iam/tenantadmin/internal/service/svctenant"
	"github.com/morehao/golib/biz/gcontext/gincontext"
)

type TenantMenuCtr interface {
	Tree(ctx *gin.Context)
}

type tenantMenuCtr struct {
	tenantMenuSvc svctenant.TenantMenuSvc
}

var _ TenantMenuCtr = (*tenantMenuCtr)(nil)

func NewTenantMenuCtr() TenantMenuCtr {
	return &tenantMenuCtr{
		tenantMenuSvc: svctenant.NewTenantMenuSvc(),
	}
}

// @Tags 租户菜单
// @Summary 当前租户菜单树
// @accept application/json
// @Produce application/json
// @Success 200 {object} gincontext.DtoRender{data=dtotenant.MenuTreeResp}
// @Router /v1/tenant/menus/tree [get]
func (ctr *tenantMenuCtr) Tree(ctx *gin.Context) {
	res, err := ctr.tenantMenuSvc.Tree(ctx)
	if err != nil {
		gincontext.Fail(ctx, err)
		return
	}
	gincontext.Success(ctx, res)
}
